acceptance/upgrade/upgrade_bound_test.go

```diff
@@ -0,0 +1,173 @@
+		appName = catalog.NewAppName()
+		service = catalog.NewServiceName()
+	})
+
+	AfterEach(func() {
+		env.DeleteNamespace(namespace)
+	})
+
+	It("Can upgrade epinio bound to a custom service", func() {
+
+		// Note current versions of client and server
+		By("Versions before upgrade")
+		env.Versions()
+
+		// Create a service
+		By("Creating a service")
+		out, err := env.Epinio("", "service", "create", "mysql-dev", service)
+		Expect(err).ToNot(HaveOccurred(), out)
+
+		// Debug only
+		fmt.Fprintf(GinkgoWriter, "Service create log: %v\n", out)
+
+		By("Wait for deployment")
+		// Make outValue global variable as we'll need it later
+		var outValue string
+		Eventually(func() string {
+			outValue, _ = env.Epinio("", "service", "show", service)
+			return outValue
+		}, "2m", "5s").Should(
+			HaveATable(
+				WithHeaders("KEY", "VALUE"),
+				WithRow("Status", "deployed"),
+			),
+		)
+		// Store the service route
+		svcRouteRegexp := regexp.MustCompile(`\b(\w{29}-mysql)`)
+		svcRoute := string(svcRouteRegexp.Find([]byte(outValue)))
+		fmt.Fprintf(GinkgoWriter, "The service table content: %v\n", outValue)
+		fmt.Fprintf(GinkgoWriter, "The service route is: %v\n", svcRoute)
+
+		// Deploy Wordpress application
+		By("Pushing Wordpress App")
+		wordpress := "https://github.com/epinio/example-wordpress,main"
+
+		pushLog, err := env.EpinioPush("",
+			appName,
+			"--name", appName,
+			"--git", wordpress,
+			"-e", "BP_PHP_WEB_DIR=wordpress",
+			"-e", "BP_PHP_VERSION=8.0.x",
+			"-e", "BP_PHP_SERVER=nginx",
+			"-e", "DB_HOST="+svcRoute,
+			"-e", "SERVICE_NAME="+service)
+		Expect(err).ToNot(HaveOccurred(), pushLog)
+
+		routeRegexp := regexp.MustCompile(`https:\/\/.*omg.howdoi.website`)
+		route := string(routeRegexp.Find([]byte(pushLog)))
+
+		Eventually(func() string {
+			out, err := env.Epinio("", "app", "list")
+			Expect(err).ToNot(HaveOccurred(), out)
+			return out
+		}, "5m").Should(
+			HaveATable(
+				WithHeaders("NAME", "CREATED", "STATUS", "ROUTES", "CONFIGURATIONS", "STATUS DETAILS"),
+				WithRow(appName, WithDate(), "1/1", appName+".*", "", ""),
+			),
+		)
+
+		By("Bind it")
+		out, err = env.Epinio("", "service", "bind", service, appName)
+		Expect(err).ToNot(HaveOccurred(), out)
+
+		By("Verify binding")
+		appShowOut, err := env.Epinio("", "app", "show", appName)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(appShowOut).To(
+			HaveATable(
+				WithHeaders("KEY", "VALUE"),
+				WithRow("Bound Configurations", svcRoute),
+			),
+		)
+
+		// Check that the app is reachable and expected page tile is reached
+		Eventually(func() int {
+			resp, err := env.Curl("GET", route, nil)
+			Expect(err).ToNot(HaveOccurred())
+			defer resp.Body.Close()
+
+			bodyBytes, err := io.ReadAll(resp.Body)
+			Expect(err).ToNot(HaveOccurred())
+
+			if resp.StatusCode == http.StatusOK && strings.Contains(string(bodyBytes), "WordPress") {
+				return resp.StatusCode
+			}
+
+			return 0
+		}, "1m", "5s").Should(Equal(http.StatusOK))
```

Comment:
This means that if the app is reacheble but it doesn't contains the "WordPress" world it will keep looping until the timeout.
I would separate the check from the "expectation":

```suggestion
		// Check that the app is reachable
		Eventually(func() int {
			resp, err := env.Curl("GET", route, nil)
			Expect(err).ToNot(HaveOccurred())
			return resp.StatusCode
		}, "1m", "5s").Should(Equal(http.StatusOK))

		// app available, check that the body contains "WordPress"
		resp, err := env.Curl("GET", route, nil)
		Expect(err).ToNot(HaveOccurred())
		bodyBytes, err := io.ReadAll(resp.Body)
		Expect(err).ToNot(HaveOccurred())
		Expect(string(bodyBytes)).To(ContainSubstring("WordPress"))
```